src/app/codegen-data-model/CodegenDataModel_Read.cpp

```diff
@@ -90,7 +89,12 @@ struct ShortPascalString
     using LengthType                        = uint8_t;
     static constexpr LengthType kNullLength = 0xFF;
 
-    static LengthType GetLength(const uint8_t * buffer) { return emberAfStringLength(buffer); }
+    static LengthType GetLength(const uint8_t * buffer)
+    {
+        // NOTE: we do NOT use emberAfLongStringLength because that will result in 0 length
+        //       for null strings
+        return *buffer;
```

Comment:
Suggest `return static_cast<LengthType>(buffer[0]);`, it's much clearer as to what is being done.

Reply:
This is specific to exactly 1 byte in the class. You are converting a 7-char `*buffer` into a 34-char that is equivalent. I don't see the point.

Reply:
used buffer[0] since I converted things to ByteSpan. I do not see why we need static_cast.